### PR TITLE
Add tutorial: "Install NixOS and partition the disk with `disko`"

### LINKED ISSUE #15: Write NixOS install tutorial (manual + disko)
Manual (not graphical) install + disko + nixos-install --flake

### PR DESCRIPTION
**Pull request by shivaraj-bh**
Resolves #15 

Local preview:
```bash
nix --refresh run github:nixos-asia/website/nixos-install-manual-disko#preview \
  -- -o /en/tutorial/nixos-install-with-disko
```

**Side note:** I am linking to some parts of [tutorial-1 of this series](https://nixos.asia/en/tutorial/nixos-install) with complete URL, something like [this](https://nixos.zulipchat.com/#narrow/stream/416826-website/topic/Link.20to.20a.20sub-section.20of.20an.20internal.20page/near/408529095) will be good to have.

### CODE REVIEW

**Comment by srid** [comment]
@shivaraj-bh The local preview command is wrong. It should be 

```sh
nix --refresh run github:nixos-asia/website/nixos-install-manual-disko#preview \
  -- -o /en/tutorial/nixos-install-with-disko
```

File: en/tutorial/nixos-install-with-disko.md [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+{#partition}
+## Partition the disk
+
+The [disko quickstart guide](https://github.com/nix-community/disko/blob/master/docs/quickstart.md) does an excellent job of explaining it. We will follow the same steps and include screenshots wherever necessary. Additionally, in the last step we will use flakes to manage the configuration.
+
+### Choosing the disk configuration
+
+Disko provides a few examples to choose [from](https://github.com/nix-community/disko/tree/master/example). We will use the [hybrid](https://github.com/nix-community/disko/blob/master/example/hybrid.nix) example as it will work for both BIOS and UEFI systems.
+
+Copy the disk configuration on to the USB flash drive.
```
Comment (srid) [inline en/tutorial/nixos-install-with-disko.md:32]: Are we really copying to this the USB flash drive?

File: en/tutorial/nixos-install-with-disko.md [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+:::
+
+Add the disk configuration and use GRUB:
+
+```nix
+{
+  # In `configuration.nix`
+  imports = [
+    ./hardware-configuration.nix
+    ./disko-config.nix
```
Comment (srid) [inline en/tutorial/nixos-install-with-disko.md:120]: Doesn't this file exist only in `/tmp/disko-config.nix`? There appears to be a step missing telling the user to move it to a suitable place.

**Review by srid — COMMENTED** [review]
I'll give this a try on my local VM.

File: en/tutorial/nixos-install-with-disko.md (hunk 1 above)
Comment (shivaraj-bh) [reply en/tutorial/nixos-install-with-disko.md:32]: Yes, the `/tmp/disko-config.nix` exists on USB flash drive, if it didn't, the config wouldn't persist on partitioning.

File: en/tutorial/nixos-install-with-disko.md (hunk 2 above)
Comment (shivaraj-bh) [reply en/tutorial/nixos-install-with-disko.md:120]: Ah, yes! will add that

File: en/tutorial/nixos-install-with-disko.md (hunk 1 above)
Comment (shivaraj-bh) [reply en/tutorial/nixos-install-with-disko.md:32]: Referring to as `/tmp` seems to be more appropriate

**Comment by shivaraj-bh** [comment]
Also heads up:

I haven't tried this yet:
```bash
sudo nixos-install --root /mnt --flake '.#nixos'
```

What I have tried is:
```bash
cd /mnt/etc/nixos
sudo nixos-install --flake '.#nixos'
```

I will try the above one on a fresh install.

**Comment by shivaraj-bh** [comment]
Also heads up:

I haven't tried this yet:
```bash
sudo nixos-install --root /mnt --flake '.#nixos'
```

What I have tried is:
```bash
cd /mnt/etc/nixos
sudo nixos-install --flake '.#nixos'
```

I will try the above one on a fresh install.

**Review by srid — COMMENTED** [review]
A couple more comments:

- Let's put images in a sub directory. Like eef9c66a921bad7978184c100e44513dda73763a
- You can now embed the "Flakeify" section in the 2nd tutorial. See 3ea59d3ceea05b21a0b20f5c78bbede5393e1cae

**Comment by srid** [comment]
@shivaraj-bh Is this post ready for me to be reviewed and edited?

**Comment by shivaraj-bh** [comment]
> @shivaraj-bh Is this post ready for me to be reviewed and edited?

I am thinking of editing the post to be more like how I talk, I will do that tonight.

Aside: I have added video walkthrough after editing some unnecessary parts and increasing the pace of the video by default. If we are going to stick with having the clip here, I can later timestamp the video and come up with an appropriate title. We can also move this to a dedicated channel of `nixos-asia`.

**Review by srid — APPROVED** [review]
Confirmed this to work in a VM. 👍

I'll make some edits. The flake page embed is inaccurate, because the user should be editing under `/mnt` not `/`. I'll try to fix this.

Also, we should write a 3rd tutorial allowing installation directly from a remote flake. This will streamline the install further. In the disko tutorial, as an user, I found that I had to do a bunch of flake editing, all of which could be avoided with an example flake to begin with. Also, desktop is not setup by the installation following this tutorial - something we can do so from the example flake. So the 3rd installation tutorial will certainly be interesting, and something we would ourselves use to setup various Juspay machines. 

Good work, @shivaraj-bh

**Comment by srid** [comment]
I think this is ready to be merged; will do today.

**Pull request merged**